numalogic/registry/dynamodb_registry.py

```diff
@@ -229,10 +233,11 @@ def load(
             # Version is provided
             v_sort_key = self._construct_version_key(sort_key, version=version)
 
-        cached_artifact = self._load_from_cache(f"{part_key}::{v_sort_key}")
+        cache_key = self._get_cache_key(part_key, v_sort_key)
+        cached_artifact = self._load_from_cache(cache_key)
 
         if cached_artifact:
-            _LOGGER.debug("Found cached artifact for key: %s", f"{part_key}::{v_sort_key}")
+            _LOGGER.debug(f"Found cached artifact for key: {cache_key}")
```

Comment:
Lets use printf style formatting for logging: https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/